sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -2779,6 +2779,23 @@ class DatasetSuite extends QueryTest
     }
   }
 
+  test("SPARK-51312: createDataFrame should work with both Date and LocalDate") {
+    val testCases = Seq(
+      ("true", Array(Row(java.time.LocalDate.of(2020, 5, 13)),
+        Row(java.time.LocalDate.of(2020, 5, 13)))),
```

Comment:
let's write the test this way:
- the input row has two columns: one is `LocalDate` and one is `Date`
- we run the test twice with the java8 datetime api on and off.

Reply:
oh it's already this way, but why do we need the `testCases`? The expected result is simply `sql("SELECT DATE'2020-05-13', ...")`.